crowdship-app/app/deliveryrequest.tsx

```diff
@@ -181,11 +343,129 @@ export default function DeliveryRequest() {
+        visible={isModalVisible}
+        transparent={true}
+        animationType="slide"
+        onRequestClose={() => setIsModalVisible(false)}
+      >
+        <View style={styles.modalOverlay}>
+          <View style={styles.modalContainer}>
+            <Text style={styles.modalTitle}>Estimate Delivery Price</Text>
+
+            {/* Add a subtitle or instruction for better user understanding */}
+            <Text style={styles.modalSubtitle}>
+              Enter the dimensions and weight of your package to estimate the delivery price.
+            </Text>
+
+            <View style={styles.dimensionsInputContainer}>
+              <View style={styles.dimensionInputWrapper}>
+                <Text>L:</Text>
+                <TextInput
+                  style={styles.dimensionInput}
+                  placeholder="Length (ft)"
+                  value={length}
+                  onChangeText={setLength}
+                  keyboardType="numeric"
+                />
+              </View>
+              <View style={styles.dimensionInputWrapper}>
+                <Text>W:</Text>
+                <TextInput
+                  style={styles.dimensionInput}
+                  placeholder="Width (ft)"
+                  value={width}
+                  onChangeText={setWidth}
+                  keyboardType="numeric"
+                />
+              </View>
+              <View style={styles.dimensionInputWrapper}>
+                <Text>H:</Text>
+                <TextInput
+                  style={styles.dimensionInput}
+                  placeholder="Height (ft)"
+                  value={height}
+                  onChangeText={setHeight}
+                  keyboardType="numeric"
+                />
+              </View>
+            </View>
+
+            <Text style={styles.modalTitle}>Enter Package Weight</Text>
+            <TextInput
+              style={styles.modalInput}
+              placeholder="Weight (kg)"
+              value={weight}
+              onChangeText={setWeight}
+              keyboardType="numeric"
+            />
+
+            <TouchableOpacity
+              style={buttonStyles.primaryButton}
+              onPress={calculatePrice}
+            >
+              <Text style={buttonStyles.buttonText}>Calculate Estimated Price</Text>
+            </TouchableOpacity>
+            <TouchableOpacity
+              style={buttonStyles.secondaryButton}
+              onPress={() => setIsModalVisible(false)}
+            >
+              <Text style={buttonStyles.buttonText}>Cancel</Text>
+            </TouchableOpacity>
+          </View>
+        </View>
+      </Modal>
     </ScrollView>
   );
 }
 
 const styles = StyleSheet.create({
+  modalOverlay: {
+    flex: 1,
+    justifyContent: "center",
+    alignItems: "center",
+    backgroundColor: "rgba(0,0,0,0.5)",
+  },
+  modalContainer: {
+    width: "80%",
+    padding: 20,
+    backgroundColor: "white",
+    borderRadius: 10,
+  },
+  modalTitle: {
```

Comment:
To keep thing consistent and reduce redundancy, use fontStyles for all text related things. You can list styles if you need to fix the margins, alignment, etc.